spec/controllers/equipment_models_controller_spec.rb

```diff
@@ -1,421 +1,275 @@
-        expect(assigns(:equipment_model).category).to be_nil
-      end
       it 'sets category when one is passed through params' do
-        cat = model.category
-        get :new, category_id: cat
-        expect(assigns(:equipment_model).category).to eq(cat)
+        cat = CategoryMock.new(traits: [:findable])
+        allow(EquipmentModel).to receive(:new)
+        get :new, category_id: cat.id
+        expect(EquipmentModel).to have_received(:new).with(category: cat)
       end
     end
-    context 'with non-admin user' do
-      before { sign_in FactoryGirl.create(:user) }
-      it 'should redirect to root' do
+    context 'when not admin' do
+      before do
+        mock_user_sign_in
         get :new
-        expect(response).to redirect_to(root_url)
       end
+      it_behaves_like 'redirected request'
     end
   end
 
-  describe 'GET edit' do
+  describe 'POST create' do
     context 'with admin user' do
-      before do
-        sign_in FactoryGirl.create(:admin)
-        get :edit, id: model
+      before { mock_user_sign_in(UserMock.new(:admin)) }
+      context 'successful save' do
+        let!(:model) { FactoryGirl.build_stubbed(:equipment_model) }
+        before do
+          allow(EquipmentModel).to receive(:new).and_return(model)
+          allow(model).to receive(:save).and_return(true)
+          post :create, equipment_model: { name: 'Model' }
+        end
+        it { is_expected.to set_flash[:notice] }
+        it { is_expected.to redirect_to(model) }
       end
-      it { is_expected.to respond_with(:success) }
-      it { is_expected.to render_template(:edit) }
-      it { is_expected.not_to set_flash }
-      it 'sets @equipment_model to selected model' do
-        expect(assigns(:equipment_model)).to eq(model)
+
+      context 'unsuccessful save' do
+        before do
+          model = EquipmentModelMock.new(save: false)
+          allow(EquipmentModel).to receive(:new).and_return(model)
+          post :create, equipment_model: { id: model.id }
+        end
+        it { is_expected.to set_flash[:error] }
+        it { is_expected.to render_template(:new) }
       end
     end
-    context 'with non-admin user' do
-      before { sign_in FactoryGirl.create(:user) }
-      it 'should redirect to root' do
-        get :edit, id: model
-        expect(response).to redirect_to(root_url)
+
+    context 'when not admin' do
+      before do
+        mock_user_sign_in
+        post :create, equipment_model:
+          FactoryGirl.attributes_for(:equipment_model)
       end
+      it_behaves_like 'redirected request'
     end
   end
 
-  describe 'POST create' do
+  describe 'PUT update' do
     context 'with admin user' do
-      before { sign_in FactoryGirl.create(:admin) }
-      context 'with valid attributes' do
+      before { mock_user_sign_in(UserMock.new(:admin)) }
+      context 'successful update' do
+        let!(:model) { FactoryGirl.build_stubbed(:equipment_model) }
```

Comment:
So here we need to use a `build_stubbed` object so that we can redirect to the updated object, right? This is in contrast to the failed update spec where we redirect to `edit` and therefore can use a simpler mock for the model. (Just thinking out loud, but correct me if I'm wrong)